assignment04/src/main/java/com/cnu2016/controller/OrdersController.java

```diff
@@ -0,0 +1,221 @@
+            if(product == null)
+                return ResponseEntity.status(HttpStatus.NOT_FOUND).body("");
+            m.setProducts(product);
+            m.setPrice(product.getSellPrice());
+            m.setQuantity(quantity);
+            mediumRepository.save(m);
+            o.setStatus("Shopping Cart");
+            ordersRepository.save(o);
+            return ResponseEntity.status(HttpStatus.CREATED).body(m);
+        }
+
+
+    }
+
+    /* Submit an Order
+        if order not found -> 404
+        Change status to "Shipping"
+        Reduce inventory
+     */
+    @Transactional
+    @RequestMapping(value = "/api/orders/{pk}", method = RequestMethod.PATCH)
+    public ResponseEntity submitOrder(@RequestBody Map<String, String> inputs, @PathVariable int pk) {
+        Orders o = ordersRepository.findOne(pk);
+        if(o == null) {                   // Order not found.
+            Map<String, String> detailObject = new HashMap<String, String>();
+            detailObject.put("detail", "Not found.");
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(detailObject);
+        } else {
+            if(o.getIsAvailable() == 0)
+                return ResponseEntity.status(HttpStatus.NOT_FOUND).body("");
+            String addressLine = inputs.get("address");
+            if(addressLine == null) {                  // Check if address if given or not.
+                Map<String, String> detailObject = new HashMap<String, String>();
+                detailObject.put("address", "Not found.");
+                return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(detailObject);
+            }
+
+            String customerName = inputs.get("user_name");
+            if(customerName != null) {     // user_name given
+                System.out.println(customerName);
+                Customer c = customersRepository.findByCustomerName(customerName);
+                if(c == null) {
+                    Customer customer = new Customer();
+                    customer.setAddressLine1(addressLine);
+                    customer.setCustomerName(customerName);
+                    o.setCustomer(customer);
+                    customersRepository.save(customer);
+                } else {
+                    o.setCustomer(c);
+                    c.setAddressLine1(addressLine);
+                    customersRepository.save(c);
+                }
+            }
+            List<Medium> m = mediumRepository.findByOrders(o);  // edit all orders.
+            int flag = 0;                       // for reverting back the quantities that are changed, if order cancelled
+            for (Medium medium : m) {
+                Product p = medium.getProducts();
+                System.out.println(p);
+                Double oldQuantity = p.getQty();
+                if(oldQuantity - medium.getQuantity() < 0) {    // check if bad request -> Quantity > available.
+                    flag = 1;
```

Comment:
You are setting flag here, and in next line checking if (flag), sending of HTTP_BAD_REQUEST can be done here.